File: AgentieTursimCS/Service/AngajatService.cs
#nullable enable
using System;
using System.Collections.Generic;
using AgentieTurismCS.Domain;
using AgentieTurismCS.Repository.Interfaces;

namespace AgentieTurismCS.Service
{
    public class AngajatService
    {
        private readonly IRepoAngajat _repoAngajat;
        
        public AngajatService(IRepoAngajat repoAngajat)
        {
            _repoAngajat = repoAngajat;
        }

        public Angajat? FindAngajatByUsername(string username)
        {
            return _repoAngajat.FindAngajatByUsername(username);
        }
        
        public Angajat? FindAngajat(Guid guid)
        {
            return _repoAngajat.FindOne(guid);
        }
        
        public IEnumerable<Angajat> FindAllAngajati()
        {
            return _repoAngajat.FindAll();
        } 
        
        public Angajat? AddAngajat(Angajat angajat)
        {
            return _repoAngajat.Save(angajat);
        }
        
        public Angajat? UpdateAngajat(Angajat angajat)
        {
            return _repoAngajat.Update(angajat);
        }
        
        public Angajat? DeleteAngajat(Angajat angajat)
        {
            return _repoAngajat.Delete(angajat.Id);
        }
    }
}

File: AgentieTurismJava/src/main/java/org/example/agentietursimjava/Service/AngajatService.java
package org.example.agentietursimjava.Service;

import org.example.agentietursimjava.Domain.Angajat;
import org.example.agentietursimjava.Repository.DBRepos.AngajatDBRepo;
import org.example.agentietursimjava.Repository.Interfaces.RepoAngajat;

import java.util.Optional;
import java.util.UUID;

public class AngajatService {
    private final RepoAngajat angajatDBRepo;

    public AngajatService(AngajatDBRepo angajatDBRepo) {
        this.angajatDBRepo = angajatDBRepo;
    }

    public Optional<Angajat> findAngajatByUsername(String username) {
        try {
            return angajatDBRepo.findOneByUsername(username);
        } catch (RuntimeException e) {
            throw new RuntimeException(e);
        }
    }
    public Optional<Angajat> findAngajat(UUID uuid) {
        try {
            return angajatDBRepo.findOne(uuid);
        } catch (RuntimeException e) {
            throw new RuntimeException(e);
        }
    }

    public Iterable<Angajat> findAllAngajati() {
        try {
            return angajatDBRepo.findAll();
        } catch (RuntimeException e) {
            throw new RuntimeException(e);
        }
    }

    public Optional<Angajat> addAngajat(Angajat angajat) {
        try {
            return angajatDBRepo.save(angajat);
        } catch (RuntimeException e) {
            throw new RuntimeException(e);
        }
    }

    public Optional<Angajat> deleteAngajat(Angajat angajat) {
        try {
            return angajatDBRepo.delete(angajat.getId());
        } catch (RuntimeException e) {
            throw new RuntimeException(e);
        }
    }

    public Optional<Angajat> updateAngajat(Angajat angajat) {
        try {
            return angajatDBRepo.update(angajat);
        } catch (RuntimeException e) {
            throw new RuntimeException(e);
        }
    }

}
